# Part 1 - Finding increasing depths
Start by opening the file, and cleaning the newline characters. Cleaning the newlines is required to avoid an off by one error when comparing the last value, which doesn't have a newline termination.

In [27]:
with open('day1_input.txt') as file:
    depth_lines = file.readlines()

depths = []
    
for line in depth_lines:
    depths.append(int(line.strip()))


Set a counter to check the number of increased values, and loop over the length of the list (minus 1 to avoid an off by 1 error). Some commented print values were used for debugging.

In [30]:
increased = 0

for i in range(len(depths)-1):
    if depths[i+1] > depths[i]:
        increased += 1
        #print(f"{depths[i]} increased to {depths[i+1]}")
    #else:
        #print(f"{depths[i]} decreased to {depths[i+1]}")
        
print(increased)

1342


The number of lines where the depth increased is 1342. This is the answer to the first part. 

# Part 2 - Using a rolling average

In order to form each rolling average, we can bump in the next value while dropping each value out the back as the new one is added. I'll make a 3 element list to hold the rolling average, and then copy each value along to the next place, making a new list of averaged depths

In [35]:
averaged_depths = []

rolling_average = [0,0,0]

for i in range(len(depths)-2):
    rolling_average[0] = depths[i]
    rolling_average[1] = depths[i+1]
    rolling_average[2] = depths[i+2]
    
    average = sum(rolling_average)/3.0
    
    averaged_depths.append(average)

Now the depths have been massaged to rolling averages, and it's easy to just reuse the same code from above

In [36]:
increased = 0

for i in range(len(averaged_depths)-1):
    if averaged_depths[i+1] > averaged_depths[i]:
        increased += 1
        
print(increased)

1378


Which is the right answer for the rolling average increase